### ISSUE TITLE
Template Path Auto Resolve

### ISSUE THREAD

**Issue by ghost** [go-aah/aah#50]
**Template path confusion?**

How do I implement a custom template file path?
```
views\pages\backend\app\index.html
views\pages\frontend\app\index.html
```

**Problem**
`app/controllers` ->
Package: frontend
Controller: App
Action: Login

**Incorrect:** It resolves to template `views/pages/app/login.html`

**Correct:** It suppose to be `views/pages/frontend/app/login.html`

----
When I call `ctx.Reply().HTMLf("index.html", data)`

**Incorrect:** It resolves to template `views/pages/app/index.html`

**Correct:** It suppose to be `views/pages/frontend/app/index.html`

**Comment by LeSou** [comment]
@jeevatkm   `	
controllerName := ctx.controller
	if strings.HasSuffix(controllerName, controllerNameSuffix) {
		controllerName = controllerName[:len(controllerName)-controllerNameSuffixLen]
	}

	tmplPath := filepath.Join("pages", controllerName)`

**Comment by jeevatkm** [comment]
My best guess is framework is resolving the template by controller name only. But it should resolve along with package of controller.

I will analyze and come back. 

If you have any error info from log. Please post it here.

**Comment by jeevatkm** [comment]
@LeSou I found the problem.

Namespace or package name is not getting added in the `tmplPath`. I will fix it. Thanks for finding an bug.

----
While I'm analyzing this issue I thought of adding this support `template path override`. 

For example: User provides the template path, framework will use the provide template path.

```go
Reply().HTMLf("/admin/index.html", data)

// resolve to template -> views/pages/admin/index.html
```

**Comment by jeevatkm** [comment]
I have created a patch for you, please apply and test it. let me know the feedback.

Steps to apply: 

Download attached patch file [template-auto-resolve.txt](https://github.com/go-aah/aah/files/1030296/template-auto-resolve.txt).
```bash
# go to directory
cd $GOPATH/src/aahframework.org/aah.v0

# run this
git apply --ignore-space-change --ignore-whitespace template-auto-resolve.txt
```

**Comment by LeSou** [comment]
solve, thank you

**Comment by jeevatkm** [comment]
@LeSou I'm reopening it. Since fix is not yet released. That patch is just for you so you can continue on your project development.

This fix will be release with `v0.6`.

**Comment by jeevatkm** [comment]
Done
